From b66bb4cbcfa69889f5b2112073cbb5c790919d8f Mon Sep 17 00:00:00 2001
From: "kirk.wang" <kirk.wang@amlogic.com>
Date: Fri, 21 Apr 2023 10:14:11 +0800
Subject: [PATCH] spandsp: fetch source fail [1/1]

PD#SWPL-120582

Problem:
fetch spandsp source fail

Solution:
modify compilation rules to remove dependencies

Verify:
t7c

Change-Id: Iabba4a59f24c25cbb20c06613fde7999923790fe
Signed-off-by: kirk.wang <kirk.wang@amlogic.com>
---
 recipes-multimedia/bluez-alsa/bluez-alsa.bb | 49 +++++++++++++--------
 1 file changed, 30 insertions(+), 19 deletions(-)

diff --git a/recipes-multimedia/bluez-alsa/bluez-alsa.bb b/recipes-multimedia/bluez-alsa/bluez-alsa.bb
index 91be0a9..7a485c2 100644
--- a/recipes-multimedia/bluez-alsa/bluez-alsa.bb
+++ b/recipes-multimedia/bluez-alsa/bluez-alsa.bb
@@ -1,13 +1,13 @@
 SUMMARY = "Bluetooth Audio ALSA Backend"
 HOMEPAGE = "https://github.com/Arkq/bluez-alsa"
-SECTION = "devel"
+SECTION = "libs"
 
 LICENSE = "MIT"
 LIC_FILES_CHKSUM = "file://LICENSE.txt;md5=bb3e99e80c5d718213f35ae1def4c106"
 
-DEPENDS = "alsa-lib bluez5 dbus glib-2.0 sbc spandsp systemd"
-
-SRC_URI = "git://github.com/Arkq/bluez-alsa.git;branch=master;protocol=https"
+SRC_URI = " \
+    git://github.com/Arkq/bluez-alsa.git;protocol=https;branch=master \
+"
 SRCREV = "9045edb436ea755f395a2e09e4525b5defad286a"
 
 SRC_URI += "file://0001-BT-a2dp-sink-add-aml-hal-audio-support-1-1.patch"
@@ -17,31 +17,42 @@ SRC_URI += "file://0001-Allow-user-to-enable-profiles-selectively.patch"
 SRC_URI += "file://0001-BT-fix-a2dp_ctl-scan-crash-in-64bit-env-1-1.patch"
 SRC_URI += "file://bluez-alsa.sh"
 
-S = "${WORKDIR}/git"
 
-inherit systemd pkgconfig autotools
+S  = "${WORKDIR}/git"
 
 DEPENDS += "alsa-lib bluez5 systemd glib-2.0 sbc"
 DEPENDS += "aml-audio-service"
-RDEPENDS_${PN} += "aml-audio-service"
+RDEPENDS:${PN} += "aml-audio-service"
 
-EXTRA_OECONF += " \
-    --enable-debug \
-    --disable-payloadcheck \
-"
+PACKAGECONFIG[debug]  = "--enable-debug"
+#PACKAGECONFIG[datadir]  = "--datadir=/etc"
+PACKAGECONFIG[aac]  = "--enable-aac, --disable-aac, "
+PACKAGECONFIG[aptx] = "--enable-aptx,--disable-aptx,"
+PACKAGECONFIG[hcitop]   = "--enable-hcitop,  --disable-hcitop,  libbsd ncurses"
+
+inherit autotools pkgconfig
+inherit systemd
 
-do_install_append () {
-    install -d ${D}${systemd_unitdir}/system
-    #install -m 0644 ${WORKDIR}/bluez-alsa.service ${D}${systemd_unitdir}/system
+SYSTEMD_AUTO_ENABLE = "enable"
+PACKAGECONFIG += "debug"
+#PACKAGECONFIG += "datadir"
+
+
+do_install:append () {
+    install -d ${D}${base_libdir}/systemd/system
     install -m 0755 ${WORKDIR}/bluez-alsa.sh ${D}${bindir}
     install -m 0755 ${B}/utils/bt-halplay ${D}${bindir}
     install -m 0755 ${B}/utils/a2dp_ctl ${D}${bindir}
     install -m 0755 ${B}/utils/hfp_ctl ${D}${bindir}
 }
 
-FILES_${PN} += "${libdir}/alsa-lib/lib*.so ${datadir}/alsa"
-FILES_${PN}-dev += "${libdir}/alsa-lib/*.la"
-FILES_${PN}-staticdev += "${libdir}/alsa-lib/lib*.a"
-FILES_${PN}-dbg += "${libdir}/alsa-lib/.debug/*.so"
+FILES:${PN} += "\
+    ${libdir}/liba2dp_ctl.so\
+    ${libdir}/alsa-lib/libasound_module_ctl_bluealsa.so\
+    ${libdir}/alsa-lib/libasound_module_pcm_bluealsa.so\
+"
 
-#SYSTEMD_SERVICE_${PN} = "bluez-alsa.service"
+FILES:${PN}-staticdev += "\
+    ${libdir}/liba2dp_ctl.a\
+    ${libdir}/alsa-lib/libasound_module_ctl_bluealsa.a\
+    ${libdir}/alsa-lib/libasound_module_pcm_bluealsa.a\
-- 
2.25.1

